src/App/App.js

```diff
@@ -2,21 +2,46 @@ import React, { Component } from 'react';
 import './App.css';
 import movieData from '../movie-data';
 import Movies from '../Movies/Movies';
+import Movie from '../Movie/Movie';
 import Header from '../Header/Header';
 
 class App extends Component {
   constructor() {
     super();
-    this.state = movieData;
+    this.state = {
+      allMovies: movieData,
+      specificMovie: null
+    };
+  }
+  displaySingleMovie = (id) => {
+    const filteredMovie = this.state.allMovies.movies.filter(movie => movie.id === id)
+    this.setState({
+      allMovies: null,
+      specificMovie: filteredMovie
+    })
+  }
+  goBackToHome = () => {
+    this.setState({
+      allMovies: movieData,
+      specificMovie: null
+    })
   }
-
   render() {
-    return(
-      <div>
-        <Header />
-        <Movies movies={this.state.movies}/>
-      </div>
-    )
+    if (!this.state.specificMovie) {
```

Comment:
We can refactor this conditional later, but for now I am so impressed how quickly we got this thing up and running!

Reply:
I'd encourage you to try to refactor your code so that you always have just one `return` rather then having `if...return, else...return` like you have here.   That said, I might not worry about that until you start to incorporate Router because that will change the game on how you're telling your app to render different things at different times/urls.